myapp/urls.py

```diff
@@ -1,12 +1,15 @@
-from django.urls import path
-from django.contrib.auth import views as auth_views
-from . import views
-
-urlpatterns = [
-	path('', views.index, name='index'),
-	path('stock/<str:symbol>/', views.single_stock, name='single_stock'),
-	path('historic/<str:symbol>/', views.single_stock_historic, name='single_stock_historic'),
-	path('accounts/login/', auth_views.LoginView.as_view(template_name='login.html'), name='login'),
-	path('accounts/logout/', views.logout_view, name='logout'),
-	path('accounts/register/', views.register, name='register'),
-]
\ No newline at end of file
+from django.urls import path, re_path
+from django.contrib.auth import views as auth_views
+from django.views.generic import TemplateView
+from . import views
+
+urlpatterns = [
+    path('', views.index, name='index'),
+    path('stock/<str:symbol>/', views.single_stock, name='single_stock'),
+    path('historic/<str:symbol>/', views.single_stock_historic, name='single_stock_historic'),
+    path('accounts/login/', auth_views.LoginView.as_view(template_name='login.html'), name='login'),
+    path('accounts/logout/', views.logout_view, name='logout'),
+    path('accounts/register/', views.register, name='register'),
+    re_path("", views.page_not_found, name='page_not_found'),
```

Comment:
the `re_path` for using custom regular expression inside the url path, while `path` can only include explicit path-names + built-in regular-expressions.
no exactly, the "`re_path`" is the new way to replace the legacy way `url` way
see this:
https://diigo.com/0io6z5

Reply:
I understand. Perhaps we can use regular expressions for all endpoints rather than hard-coded paths? That way, we would not need to do the condition if `request.path[-1] != '/':` in on line 79 of **views.py**

Reply:
I removed it, now should be better